app/scripts/modules/core/src/reactShims/modal.injector.ts

```diff
@@ -0,0 +1,18 @@
+import IInjectorService = angular.auto.IInjectorService;
+
+import { IModalService, IModalStackService } from 'angular-ui-bootstrap';
+
+import { ReactInject } from './react.injector';
+
+// prettier-ignore
+export class CoreModalInject extends ReactInject {
+  // Services
+  public get modalService(): IModalService { return this.$injector.get('$uibModal') as IModalService; }
+  public get modalStackService(): IModalStackService { return this.$injector.get('$uibModalStack') as IModalStackService; }
+
+  public initialize($injector: IInjectorService) {
+    this.$injector = $injector;
+  }
+}
+
+export const ModalInjector: CoreModalInject = new CoreModalInject();
```

Comment:
What's the thinking behind moving this to its own class outside the ReactInjector?

Reply:
When you pull in ReactInjector, it loads all the modules that it imports. This can cause a circular dependency on some of the lower level spinnaker dependencies. `ApiService --> AccountInitializer --> ReactInjector --> EntityTagReader --> ApiService`

It probably makes sense to just make one for `externals` or `vendor` and `spinnaker` modules.